nf_core/pipeline-template/{{cookiecutter.name_noslash}}/bin/check_samplesheet.py

```diff
@@ -0,0 +1,115 @@
+            os.makedirs(path)
+        except OSError as exception:
+            if exception.errno != errno.EEXIST:
+                raise
+
+
+def print_error(error,line):
+    print("ERROR: Please check samplesheet -> {}\nLine: '{}'".format(error,line.strip()))
+    sys.exit(1)
+
+
+def check_samplesheet(FileIn,FileOut):
+    ## Check header
+    HEADER = ['sample', 'fastq_1', 'fastq_2']
+    fin = open(FileIn,'r')
+    header = fin.readline().strip().split(',')
+    if header != HEADER:
+        print("ERROR: Please check samplesheet header -> {} != {}".format(','.join(header),','.join(HEADER)))
+        sys.exit(1)
+
+    sampleRunDict = {}
+    while True:
+        line = fin.readline()
+        if line:
+            lspl = [x.strip() for x in line.strip().split(',')]
+
+            ## Check valid number of columns per row
+            if len(lspl) != len(header):
+                print_error("Invalid number of columns (minimum = {})!".format(len(header)),line)
+
+            numCols = len([x for x in lspl if x])
+            if numCols < 2:
+                print_error("Invalid number of populated columns (minimum = 2)!".format(line))
+
+            ## Check sample name entries
+            sample,fastQFiles = lspl[0],lspl[1:]
+            if sample:
+                if sample.find(' ') != -1:
+                    print_error("Sample entry contains spaces!",line)
+            else:
+                print_error("Sample entry has not been specified!",line)
+
+            ## Check FastQ file extension
+            for fastq in fastQFiles:
+                if fastq:
+                    if fastq.find(' ') != -1:
+                        print_error("FastQ file contains spaces!",line)
+                    if fastq[-9:] != '.fastq.gz' and fastq[-6:] != '.fq.gz':
+                        print_error("FastQ file does not have extension '.fastq.gz' or '.fq.gz'!",line)
+
+            ## Auto-detect paired-end/single-end
+            sample_info = []                             ## [single_end, fastq_1, fastq_2]
+            fastq_1,fastq_2 = fastQFiles
+            if sample and fastq_1 and fastq_2:           ## Paired-end short reads
+                sample_info = ['0', fastq_1, fastq_2]
+            elif sample and fastq_1 and not fastq_2:     ## Single-end short reads
+                sample_info = ['1', fastq_1, fastq_2]
+            else:
+                print_error("Invalid combination of columns provided!",line)
+
+            if sample not in sampleRunDict:
+                sampleRunDict[sample] = [sample_info]
+            else:
+                if sample_info in sampleRunDict[sample]:
+                    print_error("Samplesheet contains duplicate rows!",line)
+                else:
+                    sampleRunDict[sample].append(sample_info)
+        else:
+            fin.close()
+            break
+
+    ## Write validated samplesheet with appropriate columns
+    if len(sampleRunDict) > 0:
+        OutDir = os.path.dirname(FileOut)
```

Comment:
It's Python convention for classes to start with capitals but not regular variables (hence the syntax highlighting is different).
```suggestion
        out_dir = os.path.dirname(file_out)
```

Reply:
All fixed [here](https://github.com/nf-core/tools/pull/653/commits/06e665b3922f7f2833bfe4551b5427f8ec9d2b99)